test/any/web_catalog/api-extractor-integration.es6.js

```diff
@@ -0,0 +1,395 @@
+          'writable': 1,
+        },
+      },
+      '10013': {
+        'hiddenAPI': {
+          'writable': 1,
+        },
+      },
+      '10014': {
+        'hiddenAPI': {
+          'writable': 1,
+        },
+      },
+      '10015': {
+        'extraProperty': {
+          'writable': 1,
+        },
+      },
+      '10016': {
+        'meaningfulAPI': {
+          'writable': 1,
+        },
+      },
+      '10017': {
+      },
+    },
+    'protos': {
+      '10005': '10016',
+      '10013': '10017',
+      '10014': '10004',
+    },
+    'root': 10000,
+    'types': {
+      'boolean': 2,
+      'exception': 7,
+      'null': 6,
+      'number': 3,
+      'string': 4,
+      'symbol': 5,
+      'undefined': 1,
+    },
+    'key': 'window',
+  });
+  let extractor = com.web.catalog.apiExtractor.create({});
+
+  // Integration tests.
+  describe('API extractor', function() {
+    let apiCatalog = extractor.extractWebCatalog(og);
+    it('correctly extract every first level apis for Window interface.',
+      function() {
+        expect(apiCatalog.Window.sort()).toEqual(['Function', 'Object',
+          'FunctionInterface', 'DuplicateFunctionInterface', 'ObjectInterface',
+          'ObjectLibrary', 'nonObjectLibrary', 'FunctionNonInterface',
+          'constant'].sort());
+      });
+    it('extract api for Object and Function without filter build-in apis.',
+      function() {
+        expect(apiCatalog.Object.sort()).toEqual(['arguments',
+          'constructor', 'toLocaleString', 'toString', 'valueOf'].sort());
+        expect(apiCatalog.Function.sort()).toEqual([
+          'prototype', 'caller', 'length', 'name'].sort());
+      });
+    it('only includes Function/Object with meaningful properties.', function() {
+      expect(apiCatalog.FunctionNonInterface).toBeUndefined();
+      expect(apiCatalog.FunctionInterface).toBeDefined();
+      expect(apiCatalog.nonObjectLibrary).toBeUndefined();
+      expect(apiCatalog.ObjectLibrary).toBeDefined();
+      expect(apiCatalog.ObjectInterface).toBeDefined();
+    });
+    it('filter build-in apis for non-special cases.', function() {
+      expect(apiCatalog.FunctionNonInterface).not.toContain('caller');
+      expect(apiCatalog.FunctionNonInterface).not.toContain('length');
+      expect(apiCatalog.FunctionNonInterface).not.toContain('name');
+      expect(apiCatalog.ObjectInterface).not.toContain('constructor');
+      expect(apiCatalog.ObjectInterface).not.toContain('toString');
+      expect(apiCatalog.ObjectInterface).not.toContain('toLocaleString');
+      expect(apiCatalog.ObjectInterface).not.toContain('valueOf');
+      expect(apiCatalog.ObjectLibrary).not.toContain('constructor');
+      expect(apiCatalog.ObjectLibrary).not.toContain('toString');
+      expect(apiCatalog.ObjectLibrary).not.toContain('toLocaleString');
+      expect(apiCatalog.ObjectLibrary).not.toContain('valueOf');
+    });
+    it('extract interfeaces even they are same object.', function() {
```

Comment:
What I want to say is 'it extracts separate interfaces for first level objects, even if some of them reference to the same object.'